js/ui/paygUnlockDialog.js

```diff
@@ -358,6 +362,56 @@ var PaygUnlockDialog = new Lang.Class({
         return buttonsBox;
     },
 
+    _onCodeSuccesfullyAdded(mainBox) {
+        mainBox.remove_all_children();
+
+        // successMessage will handle the formatting of the string
+        let timeString = Payg.successMessage(Main.paygManager._lastTimeAdded);
+        let successLabel = new St.Label({ style_class: 'unlock-dialog-payg-success',
+                                    text: timeString,
+                                    x_align: Clutter.ActorAlign.CENTER });
+        mainBox.add_child(successLabel);
+
+        let successButtonBox = new St.BoxLayout({ style_class: 'unlock-dialog-payg-button-box',
+                                        vertical: false,
+                                        x_expand: true,
+                                        x_align: Clutter.ActorAlign.CENTER,
+                                        y_expand: true,
+                                        y_align: Clutter.ActorAlign.END });
+
+        this._successButton = new St.Button({ style_class: 'modal-dialog-button button',
+                                         button_mask: St.ButtonMask.THREE,
+                                         reactive: true,
+                                         can_focus: true,
+                                         label: _("Success!"),
+                                         x_align: St.Align.MIDDLE,
+                                         y_align: St.Align.END });
+
+        this._successButton.add_style_pseudo_class('default');
+        successButtonBox.add_child(this._successButton);
+        mainBox.add_child(successButtonBox);
+        this._successButton.grab_key_focus();
+
+        this._timeoutId = Mainloop.timeout_add_seconds(3, () => {
+            Main.paygManager.emit('success-message-shown');
+
+            return GLib.SOURCE_REMOVE;
+        });
+
+        this._successButton.connect('button-press-event', () => {
+            this._onButtonConnect();
+        });
+
+        this._successButton.connect('key-press-event', () => {
+            this._onButtonConnect();
+        });
+    },
+
+    _onButtonConnect() {
```

Comment:
This function name is misleading. No button is doing any connecting.

Reply:
this method runs when the success button either detects a keyboard tap or a click (on a few lines above it) and it is connecting to the signal emitted by either mouse or keyboard to that object. Or did I get that wrong?